mimesis/locales.py

```diff
@@ -1,8 +1,8 @@
 import contextlib
-from typing import Generator
+from typing import TYPE_CHECKING, Generator
 
-from mimesis import providers
-from mimesis.utils import pull
+if TYPE_CHECKING:
+    from mimesis.providers.base import BaseDataProvider
```

Comment:
I would like to see a comment here, why this hack is needed. 
Something about cycle dependencies will be fine.